src/components/App.js

```diff
@@ -46,6 +49,16 @@ const App = () => {
     setTodos(updatedTodos)
   }
 
+  const handleModal = (id) => {
+    const selectedTodo = todos.find((todo) => todo.id === id)
+    setTodoModal(selectedTodo)
+  }
+
+  const handleClickDetailsTodo = (id) => {
+    setIsModalShow(!isModalShow)
```

Comment:
Al hacer click en un todo, este no "togglea" el modal, solo lo abre, entonces debería encargarse de eso solamente (principio de responsabilidad única).

 ```suggestion
    setIsModalShow(true)
```